pkg/cmd/checkadmission/admission_tester.go

```diff
@@ -0,0 +1,136 @@
+				Group:   apiextensionsv1.SchemeGroupVersion.Group,
+				Version: apiextensionsv1.SchemeGroupVersion.Version,
+				Kind:    "CustomResourceDefinition",
+			},
+			Resource: metav1.GroupVersionResource{
+				Group:    apiextensionsv1.SchemeGroupVersion.Group,
+				Version:  apiextensionsv1.SchemeGroupVersion.Version,
+				Resource: "customresourcedefinitions",
+			},
+			SubResource:        "",
+			RequestSubResource: "",
+			UserInfo:           v1.UserInfo{},
+			Object: runtime.RawExtension{
+				Raw: []byte(resourceread.WriteCustomResourceDefinitionV1OrDie(tc.ComparatorTest.NewCRD)),
+			},
+			DryRun:  nil,
+			Options: runtime.RawExtension{},
+		},
+	}
+	if tc.ComparatorTest.ExistingCRD == nil {
+		admissionReview.Request.Operation = admissionv1.Create
+	} else {
+		admissionReview.Request.Operation = admissionv1.Update
+		admissionReview.Request.OldObject = runtime.RawExtension{
+			Raw: []byte(resourceread.WriteCustomResourceDefinitionV1OrDie(tc.ComparatorTest.ExistingCRD)),
+		}
+		admissionReview.Request.Name = tc.ComparatorTest.ExistingCRD.Name
+	}
+
+	admissionBytes, err := json.Marshal(admissionReview)
+	if err != nil {
+		t.Fatal(err)
+	}
+	actual := &admissionv1.AdmissionReview{}
+	result := tc.restClient.Post().AbsPath("/apis/admission.api.openshift.io/v1/crdextendedvalidations").Body(admissionBytes).Do(ctx)
+	if body, err := result.Raw(); err != nil {
+		t.Log(string(body))
+		t.Fatal(err)
+	}
+	err = result.Into(actual)
+	if err != nil {
+		t.Fatal(err)
+	}
+	actual.Request = nil
+	//t.Log(spew.Sdump(actual))
+
+	actualResults := []manifestcomparators.ComparisonResults{}
+	actualErrors := []error{}
+	comparatorNameToResults := map[string]manifestcomparators.ComparisonResults{}
+
+	if actual.Response.Result != nil && actual.Response.Result.Details != nil {
+		for _, cause := range actual.Response.Result.Details.Causes {
+			name := string(cause.Type)
+			if name == "EvaluationError" {
+				actualErrors = append(actualErrors, fmt.Errorf(cause.Message))
+				continue
+			}
+			results := comparatorNameToResults[name]
+			results.Name = name
+			results.Errors = append(results.Errors, cause.Message)
+			comparatorNameToResults[name] = results
+		}
+	}
+
+	for _, warning := range actual.Response.Warnings {
+		if strings.HasPrefix(warning, "fyi: ") {
+			parts := strings.SplitN(warning, ":", 3)
+			name := parts[1]
+			results := comparatorNameToResults[name]
+			results.Name = name
+			results.Infos = append(results.Infos, parts[2])
+			comparatorNameToResults[name] = results
+			continue
+		}
+		parts := strings.SplitN(warning, ":", 2)
+		name := parts[0]
+		results := comparatorNameToResults[name]
+		results.Name = name
+		results.Warnings = append(results.Warnings, parts[1])
+		comparatorNameToResults[name] = results
+	}
+
+	for name := range comparatorNameToResults {
+		results := comparatorNameToResults[name]
+		actualResults = append(actualResults, results)
+	}
+
+	if len(actualResults) > 1 {
+		panic(len(actualResults))
+	}
```

Comment:
Why can there not be more than one result?

Reply:
removing.